app/questionnaire/questionnaire_schema.py

```diff
@@ -14,73 +15,79 @@ def __init__(self, questionnaire_json, language_code=DEFAULT_LANGUAGE_CODE):
         self.language_code = language_code
         self._parse_schema()
 
-    @property
-    def sections(self):
+    def get_sections(self):
         return self._sections_by_id.values()
 
-    @property
-    def groups(self):
-        return self._groups_by_id.values()
+    def get_section_ids(self):
+        return list(self._sections_by_id.keys())
 
-    @property
-    def blocks(self):
-        return self._blocks_by_id.values()
+    def get_section(self, section_id: str):
+        return self._sections_by_id.get(section_id)
 
-    @property
-    def answers(self):
-        return self._answers_by_id.values()
+    def get_section_for_block_id(self, block_id):
+        block = self.get_block(block_id)
 
-    def get_section(self, section_id):
-        return self._sections_by_id.get(section_id)
+        try:
+            group_id = block['parent_id']
```

Comment:
Personal preference I guess but just to be consistent?. Most places we don't use exception control flow. You could explicitly `return None` also but we don't in most places. 🤷‍♂ 
```python
    def get_section_for_block_id(self, block_id):
        block = self.get_block(block_id)
        group_id = block.get('parent_id')

        if group_id:
            section_id = self.get_group(group_id)['parent_id']
            if section_id:
                return self.get_section(section_id)
```

Reply:
I've refactored this method to be a lot clearer.